crates/runtime/src/datafusion.rs

```diff
@@ -1231,34 +1245,148 @@ impl DataFusion {
+                Ok(view_table) => view_table,
                 Err(e) => {
                     tracing::error!("Failed to create view: {e}");
+                    status.update_view(&table, status::ComponentStatus::Error);
                     return;
                 }
             };
 
-            let view_table = match ViewTable::try_new(plan, Some(view)) {
-                Ok(view_table) => view_table,
-                Err(e) => {
-                    tracing::error!("Failed to create view: {e}");
+            if let Some(acceleration) = acceleration {
+                if acceleration.enabled {
+                    if let Err(e) = df_ref
+                        .create_accelerated_view(
+                            &table,
+                            view_table,
+                            acceleration,
+                            &dependent_table_names,
+                            secrets,
+                        )
+                        .await
+                    {
+                        tracing::error!("Failed to create view: {e}");
+                        status.update_view(&table, status::ComponentStatus::Error);
+                    }
                     return;
                 }
-            };
+            }
+
+            // non-accelerated view
             if let Err(e) = ctx.register_table(table.clone(), Arc::new(view_table)) {
                 tracing::error!("Failed to create view: {e}");
+                status.update_view(&table, status::ComponentStatus::Error);
+                return;
             };
-
-            tracing::info!("Created view {table}");
+            tracing::info!("{}", view_registered_trace(&table, None));
+            status.update_view(&table, status::ComponentStatus::Ready);
         });
 
         Ok(())
     }
 
+    pub async fn create_accelerated_view(
+        self: &Arc<Self>,
+        table: &TableReference,
+        view_table: ViewTable,
+        acceleration: acceleration::Acceleration,
+        dependent_tables: &[TableReference],
+        secrets: Arc<TokioRwLock<Secrets>>,
+    ) -> Result<()> {
+        tracing::debug!(
+            "Creating accelerated view {table:?} with dependent tables {dependent_tables:?}"
+        );
+
+        // If accelerated view depends on other tables, wait until they are ready; this is required to complete
+        // initial data load and avoid errors indicating that the load can't be completed because tables are still loading or connecting
+
+        // Exponential retry with max duration of 10 seconds between retries
+        let retry_strategy = FibonacciBackoffBuilder::new()
+            .max_retries(None)
+            .max_duration(Some(Duration::from_secs(10)))
+            .build();
+        let runtime_status = self.runtime_status();
+        let dependent_tables = dependent_tables
+            .iter()
+            .cloned()
+            .map(resolve_table_reference)
```

Comment:
Resolving references is required as view can have tables defined as `t1`, `public.t1`, `spice.pubic.t1` and it will be different TableReference if we don't normalize them.